diag/diagnostic.go

```diff
@@ -0,0 +1,87 @@
+package diag
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/zclconf/go-cty/cty"
+)
+
+type Diagnostics []*Diagnostic
+
+func (diags Diagnostics) Append(i interface{}) Diagnostics {
+	switch v := i.(type) {
+	case Diagnostics:
+		if len(v) != 0 {
+			diags = append(diags, v...)
+		}
+	case *Diagnostic:
+		diags = append(diags, v)
+	case error:
+		if v != nil {
+			diags = append(diags, &Diagnostic{
+				Severity: Error,
+				Summary:  v.Error(),
+			})
+		}
+	}
+	return diags
+}
+
+func (diags Diagnostics) HasErrors() bool {
+	for _, d := range diags {
+		if d.Severity == Error {
+			return true
+		}
+	}
+	return false
+}
+
+func (diags Diagnostics) Error() string {
+	return multierror.ListFormatFunc(diags.Errors())
+}
```

Comment:
I really don't think this should implement the `Error` interface, people will do bad things with it.